testing/trino-tests/src/test/java/io/trino/security/TestAccessControl.java

```diff
@@ -552,6 +553,82 @@ public void testJoinBaseTableWithView()
         assertAccessAllowed(viewOwnerSession, "DROP VIEW " + viewName);
     }
 
+    @Test
+    public void testFunctionOwnersRoleGrants()
```

Comment:
I don't think it is correct testing of the functionality you have introduced. You need to check if events are propagated. It has not much to do with "ownership". It is a different layer. You need a test similar to `io.trino.execution.TestBeginQuery`. Then you don't need any changes in `TestingAccessControlManager` too.

Reply:
I was about to add another test